Mace Implementation

AzliRynz opened the pull request:
## Introduction
<!-- Explain existing problems or why this pull request is necessary -->
Implementation for the 1.21.0 `Mace`
### Relevant issues
<!-- List relevant issues here -->

N/A

### Details
https://minecraft.wiki/w/Mace

## Changes
Both class' for the Mace item have been created:

* pocketmine\item\Mace.php

One class has been added to handle mace smash ground sound:

* pocketmine/world/sound/MaceSmashGroundSound.php

And one class has been added to handle mace smash air sound:

* pocketmine/world/sound/MaceSmashAirSound.php

The Mace has been implemented as a vanilla item.

### API changes
<!-- Any additions to the API that should be documented in release notes? -->
N/A
### Behavioural changes
<!-- Any change in how the server behaves, or its performance? -->
N/A
## Backwards compatibility
<!-- Any possible backwards incompatible changes? How are they solved, or how can they be solved? -->
N/A
## Follow-up
N/A

## Mojang Screenshot
<img src="https://minecraft.wiki/images/thumb/Noor_holding_a_Mace.png/240px-Noor_holding_a_Mace.png?9dea5">
Noor holding a Mace
<img src="https://minecraft.wiki/images/thumb/Bedrock_1.20.80.22_PatchNotes.png/240px-Bedrock_1.20.80.22_PatchNotes.png?e3ec9">
Noor using the mace against a zombie.

## Video


https://github.com/user-attachments/assets/d8f636a1-8b6e-4ff1-bc0f-b7c7b26a8585



## Tests
<!--
PRs which have not been tested MUST be marked as draft.
-->
I tested this PR by doing the following (tick all that apply):
- [ ] Writing PHPUnit tests (commit these in the `tests/phpunit` folder)
- [x] Playtesting using a Minecraft client (provide screenshots or a video)
- [ ] Writing a test plugin (provide the code and sample output)
- [ ] Other (provide details)

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace

Commit:
Add method getMaxDurability for Mace Implementation

Commit:
FIX CS

Commit:
FIX CS

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
FIX CS

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
FIX CS

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
FIX CS

Commit:
Fix CS

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace

Commit:
Mavewjssj

Commit:
Mace Implementation

Commit:
Rrrrrrrr

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Implemented: MaceSmashAirSound and MaceSmashGround

Commit:
FIX CS

Commit:
FIX CS

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
FIX CS

Commit:
Create build.yml

Commit:
J

Commit:
Update build.yml

Commit:
Delete .github/workflows/build.yml

AzliRynz commented:
If there is anything that needs to be fixed, please let me know.

JavierLeon9966 commented:
1. Smash particles that increase per distance fallen
2. Damage to the entity increased per distance fallen
3. Knock back to nearby entities to the smashed entity
4. The smash sounds shouldn't be played when breaking a block or normally hitting an entity. For example, the heavy smash is played when falling too high
5. The player falling motion is reset when smashing something
6. You're using `applyDamage` wrong. That's for changing the item's durability

Commit:
Mace Implementation

JavierLeon9966 commented 2 hours ago
Smash particles that increase per distance fallen
Damage to the entity increased per distance fallen
Knock back to nearby entities to the smashed entity
The smash sounds shouldn't be played when breaking a block or normally hitting an entity. For example, the heavy smash is played when falling too high
The player falling motion is reset when smashing something
You're using applyDamage wrong. That's for changing the item's durability

Commit:
Mace Implementation.

Commit:
FIX CS

Commit:
FIX CS

Commit:
FIX CS

Commit:
Merge pull request #1 from KnosTx/KnosTx-patch-1

Mace Implementation.                     JavierLeon9966

Commit:
Mace Implementation

AzliRynz commented:
I want to test whether it works or not

Commit:
Create php.yml

Commit:
Wrong branch

Commit:
Update StringToItemParser.php

Commit:
Mace Implementation

Commit:
Remove

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Mace Implementation

Commit:
Update Mace.php

Commit:
Update Mace.php

Commit:
Mace Implementation

Commit:
Update Mace.php

Commit:
Update M

Commit:
D

Commit:
Update Mace.php

Commit:
Update Mace.php

Commit:
Update Mace.php

Commit:
Update Mace.php

Commit:
Mace Implementation
